### PR TITLE
Move maven push gradle to gradle folder

### PR DESCRIPTION
**Pull request by jaredsburrows**
- Moved `maven-push` to `gradle` folder to make it easily re-usable
- Made POM variables able to read in top `build.gradle` folder

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -33,6 +33,24 @@ ext {
     supportLibraryVersion = "24.2.1"
     butterKnifeVersion = "7.0.1"
     daggerVersion = "2.0.2"
+
+    // POM file
+    GROUP = "com.github.brianPlummer"
```
Comment (jaredsburrows) [inline build.gradle]: Easy to read, you can verify here: https://mvnrepository.com/artifact/com.github.brianPlummer/tinydancer/0.0.9 and a good example here: https://github.com/square/leakcanary/blob/050b554887cd590c80803c25dcd6c05bdf89eadc/build.gradle#L20.

File: gradle/maven-push.gradle [hunk 2]
```diff
@@ -92,21 +77,47 @@ afterEvaluate { project ->
         sign configurations.archives
     }
 
-    //task androidJavadocs(type: Javadoc) {
-    //source = android.sourceSets.main.allJava
-    //}
+    task androidJavadocs(type: Javadoc) {
```
Comment (jaredsburrows) [inline gradle/maven-push.gradle]: No reason to comment this out, this task works. :)

File: gradle/maven-push.gradle [hunk 3]
```diff
@@ -92,21 +77,47 @@ afterEvaluate { project ->
...
+        source = android.sourceSets.main.java.srcDirs
+        ext.androidJar = "${android.sdkDirectory}/platforms/${android.compileSdkVersion}/android.jar"
+        classpath += files(ext.androidJar)
+    }
 
-    //task androidJavadocsJar(type: Jar, dependsOn: androidJavadocs) {
-    //classifier = 'javadoc'
-    //from androidJavadocs.destinationDir
-    //}
+    task androidJavadocsJar(type: Jar, dependsOn: androidJavadocs) {
```
Comment (jaredsburrows) [inline gradle/maven-push.gradle]: No reason to comment this out, this task works. :)

File: gradle/maven-push.gradle [hunk 4]
```diff
@@ -92,21 +77,47 @@ afterEvaluate { project ->
...
+            }
+        }
+    }
+
     artifacts {
         archives androidSourcesJar
+        archives androidJavadocsJar
+    }
+
+    task installArchives(type: Upload) {
```
Comment (jaredsburrows) [inline gradle/maven-push.gradle]: To test out `tinydancer` locally, you can run ./gradlew installArchives to publish to your local maven folder: `.m2`.

File: build.gradle [hunk 5]
```diff
@@ -33,6 +33,24 @@ ext {
...
+    VERSION_NAME = "1.0.0-SNAPSHOT"
```
Comment (jaredsburrows) [inline build.gradle]: - `-SNAPSHOT` means it will but published to snapshots
- Removing this will publish to mavenCentral

**Comment by digitalbuddha** [comment]
👍

**Comment by digitalbuddha** [comment]
`To run dex in process, the Gradle daemon needs a larger heap.
It currently has 1024 MB.
For faster builds, increase the maximum heap size for the Gradle daemon to at least 1536 MB.
To do this set org.gradle.jvmargs=-Xmx1536M in the project gradle.properties.`
Next pr? or sneak it in?

**Comment by jaredsburrows** [comment]
@digitalbuddha Just ignore that. That is the newer plugin letting us know that we can make it faster by giving it more memory.

**Pull request merged**